src/ray/common/function_descriptor.h

```diff
@@ -0,0 +1,181 @@
+#ifndef RAY_CORE_WORKER_FUNCTION_DESCRIPTOR_H
+#define RAY_CORE_WORKER_FUNCTION_DESCRIPTOR_H
+
+#include <string>
+
+#include "ray/common/grpc_util.h"
+#include "ray/protobuf/common.pb.h"
+
+namespace ray {
+/// See `common.proto` for definition of `FunctionDescriptor` oneof type.
+using FunctionDescriptorType = rpc::FunctionDescriptor::FunctionDescriptorCase;
+/// Wrap a protobuf message.
+class FunctionDescriptorInterface : public MessageWrapper<rpc::FunctionDescriptor> {
```

Comment:
Do you need to hold a shared_ptr of rpc::FunctionDescriptor internally in this class? Otherwise it seems not necessary to inherit from MessageWrapper, which always does a copy if you don't pass it a shared_ptr

Reply:
Yes, a new FunctionDescriptor is a copy of proto message. It is similar to `TaskSpecification`. I rarely create a new FunctionDescriptor from TaskSpecification.